packages/core/src/App/Components/Routes/language-link.tsx

```diff
@@ -54,9 +54,9 @@ const LanguageLink = observer(({ icon_classname, is_clickable = false, lang, tog
                     id={`dt_settings_${lang}_button`}
                     key={lang}
                     onClick={async () => {
-                        await changeSelectedLanguage(lang);
-                        await i18n.changeLanguage?.(lang);
                         toggleModal?.();
+                        changeSelectedLanguage(lang);
```

Comment:
`changeSelectedLanguage` does not return any Promise, that's why `await` is not needed.
The main fix is preventing a user from clicking a language button by toggling the modal before waiting for a Promise, which is returned from `i18n.changeLanguage?.(lang)`, to resolve. :)